### PR TITLE
correct extreme uses of hide() and protect()

### PR DESCRIPTION
**Pull request by EvilDrW**
this addresses the problem of #254 where, if you hide or protect all fields of an object during a PUT operation, all the data of the record will be deleted.

### CODE REVIEW

File: lib/resources/collection/index.js [hunk 1]
```diff
@@ -567,6 +567,10 @@ Collection.prototype.save = function (ctx, fn) {
         }
 
         delete item.id;
+        if ((_.keys(prev).length > 0) && (_.keys(item).length === 0)) {
+          return done(null, {id: id});
```
Comment (andreialecu) [inline lib/resources/collection/index.js]: I think this should return an empty object `{}` instead of the `id`. Maybe the id is intended to be hidden.

**Comment by EvilDrW** [comment]
that is a good point.  changed to match.

**Comment by andreialecu** [comment]
Would you mind rebasing this on master so I can merge it?

Thanks!

**Comment by EvilDrW** [comment]
i'm confused... does something still need to be done?  i don't see anywhere that it says it needs to be rebased

**Comment by NicolasRitouet** [comment]
Merged, thanks for your contribution 👍

**Pull request merged**